OWL2/java/de/unibremen/informatik/OWL2Parser.java

```diff
@@ -45,20 +48,87 @@
 	private static OWLOntologyIRIMapperImpl mapper =
 		new OWLOntologyIRIMapperImpl();
 	private static URI uri = null;
+	private static int maxRedirects = 0;
+
+	private static void checkMaxRedirects() {
+		String t = System.getProperty("http.maxRedirects", null);
+		if (t == null || t.isEmpty()) {
+			maxRedirects = 20;
+			return;
+		}
+		try {
+			int n = Integer.parseInt(t, 10);
+			if (n > 0) {
+				maxRedirects = n;
+			}
+		} catch (NumberFormatException nfe) {
+			System.err.println("Ignoring invalid system property "
+				+ "'http.maxRedirects' - invalid number '" + t + "'");
+		}
+	}
+
+	private static URLConnection getConnection(URL url, int redirCount)
+		throws IOException
+	{
+		URLConnection c = url.openConnection();
+		if (!(c instanceof HttpURLConnection)) {
+			return c;
+		}
+		HttpURLConnection con = (HttpURLConnection) c;
+		con.addRequestProperty("Accept", "text/plain");
+		con.setConnectTimeout(300 * 1000);	// 5 min timeout
```

Comment:
No.

This is not related to "download time". It is related to the tcp connection (initial handshake) setup and basically says: if a connection can't be established within 5 min, stop waiting for an answer and fail. Therefore I think 5 min are more than sufficient.

Another tweak would be the readTimeout setting, i.e. how long max. a single read on the related socket may block. This would go a little bit more in the "server too busy to answer" direction (IIRC the default is "wait forever"). So setting both to lower values is more or less a resource optimization, higher values usually make not much sense.

Anyway, therefore I think, "download size/time" is kind of orthogonal to these settings.